store/tikv/2pc.go

```diff
@@ -497,19 +501,30 @@ func (c *twoPhaseCommitter) buildPrewriteRequest(batch batchKeys, txnSize uint64
 		IsPessimisticLock: isPessimisticLock,
 		ForUpdateTs:       c.forUpdateTS,
 		TxnSize:           txnSize,
+		MinCommitTs:       minCommitTS,
 	}
 	return tikvrpc.NewRequest(tikvrpc.CmdPrewrite, req, pb.Context{Priority: c.priority, SyncLog: c.syncLog})
 }
 
-func (actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoffer, batch batchKeys) error {
+func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoffer, batch batchKeys) error {
 	txnSize := uint64(c.regionTxnSize[batch.region.id])
 	// When we retry because of a region miss, we don't know the transaction size. We set the transaction size here
 	// to MaxUint64 to avoid unexpected "resolve lock lite".
 	if len(bo.errors) > 0 {
 		txnSize = math.MaxUint64
 	}
 
+	isPrimary := bytes.Equal(batch.keys[0], c.primary())
 	req := c.buildPrewriteRequest(batch, txnSize)
+
+	if x := bo.ctx.Value("checkLargeTxnEnable"); x != nil {
```

Comment:
This has some extra cost, can we avoid it?

Reply:
I tried failpoint but the concurrent test is a problem.

Reply:
The test doesn't make much sense.